nixos/modules/system/activation/top-level.nix

```diff
@@ -306,6 +313,21 @@ in
       '';
     };
 
+    system.includeBuildDependencies = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to include the build closure of the whole system in
+        its runtime closure.  This can be useful for making changes
+        fully offline, as it results in all sources and patches
+        required to build the system being included.
```

Comment:
```suggestion
        Whether to include the build closure of the whole system in
        its runtime closure.  This can be useful for making changes
        fully offline, as it includes all sources, patches and intermediate
        outputs that are required to build all the derivations.
        All derivations really means all derivations, from the bootstrapping
        tools, to the build tools, to the packages, and all the way up to the
        configuration files of the system.
```